File: Data_Structure/javascript/9012.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');
for (let i = 1,stack,result ; i<input.length-1 ; i++){
  stack = [];
  result = "YES";
  for (let j = 0 ; j<input[i].length ; j++){
    if (input[i][j] === "(") stack.push("(");
    else{
      if (stack.length === 0) {result = "NO"; break;}
      stack.pop();
    }
  }
  if (stack.length !== 0) result ="NO";
  console.log(result)
}

File: Data_Structure/python/9012.py
T = int(input())
for _ in range(T):
    s = input()
    stack = []
    result = "YES"
    for char in s:
        if char =="(":
            stack.append(char)
        else:
            if not stack:
                result = "NO"
                break
            stack.pop()
    if stack:
        result = "NO"
    print(result)
